### ISSUE TITLE
no time-sheets listed in view

### ISSUE THREAD

**Issue by hal9000hal** [dhanasingh/redmine_wktime#54]
Just upgraded from redmine 2.6 to 3.2:

Tried both the current dev and master version of wktime.

No time sheets are listed in the Time & Attendance view (There should be dozens). If a new time sheet is made, when it's being edited you can move to previous time sheets from that view. but upon save of new time sheet it does not show up in the view.

![image](https://cloud.githubusercontent.com/assets/18753077/14937104/c3009624-0ec2-11e6-802a-1b6e6aaa694b.png)

If Date range of "all time" is selected an internal error is generated, if the default project is selected (1st project in list) . But if I select a different project it does not do this.  Here is the production log from that event.

Started GET "/redmine/wktime/index?utf8=%E2%9C%93&tab=wktime&searchlist=wktime&filter_type=1&project_id=48&user_id=3&status%5B%5D=e&status%5B%5D=n&status%5B%5D=r&status%5B%5D=s&status%5B%5D=a&period_type=1&period=all" for 192.168.50.180 at 2016-04-30 11:01:01 -0500
Processing by WktimeController#index as HTML
  Parameters: {"utf8"=>"✓", "tab"=>"wktime", "searchlist"=>"wktime", "filter_type"=>"1", "project_id"=>"48", "user_id"=>"3", "status"=>["e", "n", "r", "s", "a"], "period_type"=>"1", "period"=>"all"}
  Current user: hglenn (id=3)
Completed 500 Internal Server Error in 42ms (ActiveRecord: 32.0ms)

NoMethodError (undefined method `startday' for #<TimeEntry id: nil>):
  plugins/redmine_wktime/app/controllers/wktime_controller.rb:1822:in `getAllTimeRange'
  plugins/redmine_wktime/app/controllers/wktime_controller.rb:74:in `index'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'
  plugins/redmine_dmsf/lib/redmine_dmsf/webdav/no_parse.rb:33:in `call'

Thanks in advance for any help the can be offered!

**Comment by dhanasingh** [comment]
Can you try pulling code using the tag, the current version is v2.1.2
Do you see anyone in the Members dropdown ?

**Comment by hal9000hal** [comment]
Hi, thanks for taking the time to look into this.  I've just ran through the upgrade again with the same behavior, as follows.

Yes, all the members are in the dropdown, it does not appear that any combination of member, project or group will show an existing time sheet.

I've tried on 2.1, 2.1.2, dev, and master.

I'm running a fresh upgrade from 2.6 to 3.2 redmine, and here is the output of production.log when it should be showing a list of sheets.

> Started GET "/redmine/wktime/index?utf8=%E2%9C%93&tab=wktime&searchlist=wktime&filter_type=1&project_id=40&user_id=0&status%5B%5D=e&status%5B%5D=n&status%5B%5D=r&status%5B%5D=s&status%5B%5D=a&period_type=1&period=current_month" for 192.168.50.166 at 2016-05-18 00:41:58 -0500
> Processing by WktimeController#index as HTML
>   Parameters: {"utf8"=>"✓", "tab"=>"wktime", "searchlist"=>"wktime", "filter_type"=>"1", "project_id"=>"40", "user_id"=>"0", "status"=>["e", "n", "r", "s", "a"], "period_type"=>"1", "period"=>"current_month"}
>   Current user: hglenn (id=3)
>   Rendered plugins/redmine_wktime/app/views/wktime/_tab_content.html.erb (0.0ms)
>   Rendered plugins/redmine_wktime/app/views/wktime/_tab_content.html.erb (0.0ms)
>   Rendered plugins/redmine_wktime/app/views/wktime/_tab_content.html.erb (0.0ms)
>   Rendered plugins/redmine_wktime/app/views/wktime/_tab_content.html.erb (0.0ms)
>   Rendered common/_tabs.html.erb (1.0ms)
>   Rendered plugins/redmine_wktime/app/views/wktime/_date_range.html.erb (0.8ms)
>   Rendered plugins/redmine_wktime/app/views/wktime/_te_index.html.erb (5.2ms)
>   Rendered plugins/redmine_wktime/app/views/wktime/index.html.erb within layouts/base (6.9ms)
>   Rendered plugins/clipboard_image_paste/app/views/clipboard_image_paste/_headers.html.erb (0.3ms)
>   Rendered plugins/clipboard_image_paste/app/views/clipboard_image_paste/_add_form.html.erb (1.4ms)
> Completed 200 OK in 1609ms (Views: 17.2ms | ActiveRecord: 1566.6ms)
> 

Again, if I create a new sheet, i can press the back button to go to old sheets, but when I return to the top level "time & Expense" page, there are no sheets listed. No combination of member,s projects or groups seems to list them.

Thanks!

**Comment by hal9000hal** [comment]
Ok, here should be a clue. I was running 1.7 under redmine 2.6, it also works correctly if I move to 1.8 under Redmine 3.2 (1.7 does not seem to work under redmine 3.2), but! 1.8.1 has the same problem as the current master. So whatever causes this happened between 1.8 and 1.8.1

**Comment by dhanasingh** [comment]
T&A 1.8 and 1.8.1 is compatible with Redline 3.0
T&A 2.1 is compatible with Redline 3.2
Please check the plugin compatibility at the Redline Plugin Page
https://www.redmine.org/plugins/wk-time

**Comment by hal9000hal** [comment]
Hello,
It appears I have no upgrade path, as soon as I move from 1.8 to 1.8.1 I lose the ability to view my current time sheets. 

When upgrading to the newest version or TIP under redmine 3.2 the time sheet views are not restored. But if I remain at 1.8 under redmine 3.2 I can see my old time sheets and users can continue to use the system. 

Thanks

**Comment by dhanasingh** [comment]
ok, is it possible to send the screenshots of the timesheet list page for 1.8 and 1.8.1 ?
with Redline 3.2, did you try T&A 2.1 ?

**Comment by hal9000hal** [comment]
Here is a screenshot of 1.8, if I move to 1.8.1 it looks the same down to the apply buttons, the total hours then lists 0 instead of 40.5 hours, and no sheets are listed. 

Post # 1  of this thread has an example image of when I tired 3.2 with 2.1 and TIP.  What information can I provide to you to help debug this?

Tanks.

![image](https://cloud.githubusercontent.com/assets/18753077/16199804/a5b42cc6-36d0-11e6-8c76-c53e958844a3.png)

**Comment by dhanasingh** [comment]
between the two screenshots, the status dropdown is changed to a list box, can you try selecting a particular status (for e.g., new or submitted etc)and see if it shows up

**Comment by hal9000hal** [comment]
OK, more info for you.

It doesn't seem to matter which status is selected in the list box, the result always looks the same.
After further testing, we've found that it works on 1.8.1; it actually breaks on version 1.9 and later, including the current version.  It only appeared not to work on 1.8.1 before because the default status selected on the page had changed.

We've done some debugging under 1.9 and found that when we reach this section in _te_index.html.erb:


```
<% unless @entries.empty? %>
<%= render :partial => 'wktime/list', :locals => { :entries => @entries }%>
<span class="pagination"><%= pagination_links_full @entry_pages, @entry_count %></span>
@entries is empty.
```


It looks like @entries should be populated in wktime_controller.rb, in the function findBySql.  The SQL query used to get the number of table entries has massively increased in size between 1.8.1 and 1.9 and I don't have time right now to study it in more detail, but the bottom line is that this query (from 1.8.1) results in the correct number of table entries (3) in @entry_count:


`select count(*) as id from (select v1.user_id, v1.startday as spent_on, v1.hours from (select date(t.spent_on , '-' || ((strftime('%w', t.spent_on)+7-0)%7) || ' days') as startday,  t.user_id, sum(t.hours) as hours ,max(t.id) as id from time_entries t, users u where u.id = t.user_id and u.id in (4) and t.spent_on between '2016-06-26' and '2016-08-06' group by startday, user_id order by startday desc, user_id ) as v1 left outer join wktimes w on v1.startday = w.begin_date and v1.user_id = w.user_id) as v2`

and this query from 1.9 results in 0 in @entry_count:

`
select count(*) as id from (select tmp3.user_id, tmp3.spent_on, tmp3.hours, tmp3.status, tmp3.status_updater, tmp3.created_on from (select tmp1.id as user_id, tmp1.created_on, tmp1.selected_date as spent_on, case when tmp2.hours is null then 0 else tmp2.hours end as hours, case when tmp2.status is null then 'e' else tmp2.status end as status, tmp2.status_updater from (select u.id, u.created_on, v.selected_date from (select date('2016-06-26' , '+' || t4.i*7*10000 + t3.i*7*1000 + t2.i*7*100 + t1.i*7*10 + t0.i*7 || ' days') selected_date from (select 0 i union select 1 union select 2 union select 3 union select 4 union select 5 union select 6 union select 7 union select 8 union select 9) t0, (select 0 i union select 1 union select 2 union select 3 union select 4 union select 5 union select 6 union select 7 union select 8 union select 9) t1, (select 0 i union select 1 union select 2 union select 3 union select 4 union select 5 union select 6 union select 7 union select 8 union select 9) t2, (select 0 i union select 1 union select 2 union select 3 union select 4 union select 5 union select 6 union select 7 union select 8 union select 9) t3, (select 0 i union select 1 union select 2 union select 3 union select 4 union select 5 union select 6 union select 7 union select 8 union select 9) t4) v, (select distinct u.id, u.created_on from users u) u where selected_date between '2016-06-26' and '2016-08-06') tmp1 left join (select v1.user_id, v1.startday as spent_on, v1.hours, case when w.status is null then 'n' else w.status end as status , un.firstname || ' ' || un.lastname as status_updater  from (select date(t.spent_on , '-' || ((strftime('%w', t.spent_on)+7-0)%7) || ' days') as startday,  t.user_id, sum(t.hours) as hours ,max(t.id) as id from time_entries t, users u where u.id = t.user_id and u.id in (4) and t.spent_on between '2016-06-26' and '2016-08-06' group by startday, user_id order by startday desc, user_id ) as v1 left outer join wktimes w on v1.startday = w.begin_date and v1.user_id = w.user_id left outer join users un on un.id = w.statusupdater_id) tmp2 on tmp1.id = tmp2.user_id and tmp1.selected_date = tmp2.spent_on where tmp1.id in (4) ) tmp3  left outer join (select min( date(t.spent_on , '-' || ((strftime('%w', t.spent_on)+7-0)%7) || ' days') ) as min_spent_on, t.user_id as usrid from time_entries t, users u  where u.id = t.user_id and u.id in (4) group by t.user_id ) vw on vw.usrid = tmp3.user_id WHERE ((tmp3.spent_on between case when vw.min_spent_on is null then date(date(tmp3.created_on) , '-' || ((strftime('%w', date(tmp3.created_on))+7-0)%7) || ' days') else vw.min_spent_on end and '2016-07-16') OR ((tmp3.spent_on < case when vw.min_spent_on is null then date(date(tmp3.created_on) , '-' || ((strftime('%w', date(tmp3.created_on))+7-0)%7) || ' days') else vw.min_spent_on end and tmp3.status <> 'e') OR (tmp3.spent_on > '2016-07-16' and tmp3.status <> 'e')))  order by tmp3.spent_on desc, tmp3.user_id) as v2`


We are using sqlite as our database, if that helps at all.

Thanks for the time.

**Comment by karthickmadhiazhagan** [comment]
Fixed in v2.3 and it is released.

**Comment by hal9000hal** [comment]
It worked! Thanks!
